rest/config_legacy.go

```diff
@@ -97,11 +97,17 @@ func (lc *LegacyServerConfig) ToStartupConfig() (*StartupConfig, DbConfigMap, er
 	// find a database's credentials for bootstrap (this isn't the first database config entry due to map iteration)
 	bsc := &BootstrapConfig{}
 	for _, dbConfig := range lc.Databases {
-		if dbConfig.Server == nil || *dbConfig.Server == "" {
+		server := dbConfig.Server
+		if server == nil || *server == "" {
 			continue
 		}
+		if strings.HasPrefix(*server, "http://") {
+			*server = strings.TrimPrefix(*server, "http://")
+			*server = fmt.Sprintf("couchbase://%s", *server)
+		}
```

Comment:
I think there are a few cases here that will need to be handled for the [variations of `server` that we allow to be used](https://docs.couchbase.com/sync-gateway/current/configuration-properties.html#databases-this_db-server).

- `dbConfig.Server` can contain a list of server addresses, not just one. This is so that SG can fall back if the first server it attempts is unreachable.
  - Will need to do this protocol scheme switch for each address found.
- The HTTP port needs stripping when using `couchbase://` or else you get this error from the SDK:
  - `couchbase://host:8091 not supported for couchbase:// scheme. Use couchbase://host`
  - This is because when using `couchbase(s)://`, the SDK avoids HTTP altogether and talks directly with port `11210`

It's starting to feel like we might want to make use of `gocbconnstr.Parse()` instead of string manipulation to transform the server URLs.